### PR TITLE
Feature/clockss upload

### PR DESCRIPTION
**Pull request by RK206**
Merging recent changes for Preservation upload

### CODE REVIEW

**Review by Steven-Eardley — CHANGES_REQUESTED** [review]
I may have further feedback on structure once i've seen this all together in the feature branch - currently I am wondering whether Preserve, Preservation and PreservationPackage might be structured better. For now, there's smaller changes commented in this PR.

File: portality/models/preservation.py [hunk 1]
```diff
@@ -22,6 +22,14 @@ def filename(self):
     def owner(self):
         return self.data.get("owner")
 
+    @property
+    def background_task_id(self):
+        return self.data.get("background_task_id")
+
+    @status.setter
```
[the remark this answers is not in the document]
Comment (RK206) [reply portality/models/preservation.py:29]: Good catch. Updated

File: portality/tasks/preservation.py [hunk 2]
```diff
@@ -14,7 +15,7 @@
 from portality.core import app
 from portality.decorators import write_required
 from portality.lib import dates
-from portality.models import Article, BackgroundJob
+from portality.models import Article, BackgroundJob, Preserve
```
[the remark this answers is not in the document]
Comment (RK206) [reply portality/tasks/preservation.py:18]: Yes will find a better  name for it

File: portality/tasks/preservation.py [hunk 3]
```diff
@@ -92,11 +117,11 @@ def cleanup(self):
...
         Preservation.delete_local_directory(local_dir)
 
-    def validate_response(self, response, tar_file, sha256):
+    def validate_response(self, response, tar_file, sha256, model):
         """
         Validate the response from server
         :param response: response object
-        :param tar_file: tar files
+        :param tar_file: tar file name
         :param sha256: sha256sum value
```
[the remark this answers is not in the document]
Comment (RK206) [reply portality/tasks/preservation.py:125]: Added doc string

File: portality/tasks/preservation.py [hunk 4]
```diff
@@ -224,10 +270,10 @@ def create_local_directories(self):
         to create preservation package
         """
         try:
-            os.mkdir(self.__local_dir)
-            os.mkdir(self.__preservation_dir)
+            os.makedirs(self.__local_dir, exist_ok=True)
```
[the remark this answers is not in the document]
Comment (RK206) [reply portality/tasks/preservation.py:273]: No it is not  a problem if folder exist. This is to make sure folder iis available and create if not available

**Comment by RK206** [comment]
Changes are committed. Ready for review

File: portality/tasks/preservation.py [hunk 5]
```diff
@@ -43,13 +50,14 @@ def prepare(cls, username, **kwargs):
         """
 
         created_time = dates.format(datetime.utcnow(), "%Y-%m-%d-%H-%M-%S")
-        dir_name = username + "-" + created_time
+        dir_name = "rama-2021-07-28-13-05-27" #username + "-" + created_time
```
Comment (richard-jones) [inline portality/tasks/preservation.py]: looks like some testing code slipping through here

**Review by richard-jones — CHANGES_REQUESTED** [review]
One really minor thing.  Otherwise, I haven't tested this in detail, but I will do that soon.

**Review by richard-jones — COMMENTED** [review]
There seem to be some changes in this PR that come from the background jobs refactoring that I did a while back.  Can we just check what's going on here, to make sure nothing is messed up with the PR?  In particular look at the changes in `portality/models/background.py`

**Comment by RK206** [comment]
@richard-jones  I fixed merge conflicts and clockss_main got merged into clockss_upload when I commit the changes.
I tested clockss_upload and everything looks fine

**Pull request merged**